### PR TITLE
gradle: split off python cdk

### PR DESCRIPTION
**Pull request by postamar**
This creates a new, stand-alone gradle project for the python CDK. This gradle project is still in use by the github workflow which publishes the CDK.

When we previously did `./gradlew :airbyte-cdk:python:generate` now we do `cd airbyte-cdk/python/` and `./gradlew :generate`.

This is best reviewed commit-by-commit.

### CODE REVIEW

File: airbyte-cdk/python/gradle.properties [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# NOTE: if you want to override this for your local machine, set overrides in ~/.gradle/gradle.properties
+
+org.gradle.parallel=true
+org.gradle.caching=true
+
+# Note, this might have issues on the normal Github runner.
+org.gradle.vfs.watch=true
+
+# Tune # of cores Gradle uses.
+# org.gradle.workers.max=3
```
Comment (postamar) [inline airbyte-cdk/python/gradle.properties:11]: copied from `./gradle.properties`.

**Review by stephane-airbyte — COMMENTED** [review]
can you explain the change in inline docker plugin usage in python CDK"? I feel like there's a sh file that should be removed somewhere

**Comment by postamar** [comment]
> I feel like there's a sh file that should be removed somewhere

You're right, and it will be, as a follow-up to https://github.com/airbytehq/airbyte/pull/35307

> can you explain the change in inline docker plugin usage in python CDK"? 

Sure. The `airbyte-docker-legacy` plugin is the descendant of what we used pre-airbyte-ci to generate docker images as part of the gradle build. As such, it caters to a wide variety of use cases, most of which are no longer relevant, especially of `base-normalization` is removed. What it does is manage efficient (i.e. cached) delegation to `tools/bin/build_image.sh`, a shell script which essentially wraps a `docker build` command.

In this particular, remaining use case, efficiency is not a concern, building the image in question is cheap even with a cold docker, plus the image isn't built as part of the regular build tasks. Hence why it boils down to a new shell script with just a few lines.

**Review by stephane-airbyte — APPROVED** [review]

**Review by girarda — APPROVED** [review]
change looks good. I tested the gradle build command and confirmed it works. I haven't tested the updated GHA

**Comment by natikgadzhi** [comment]
Unrelated to this PR: @girarda, over the long term, please tell me we can use Python tooling to build and publish the CDK! We just talked with @alafanechere that we'd like to move the CDK to Poetry, perhaps that's a good moment to migrate the CDK on the same flow of publishing the pypi package that we use for connectors?

**Comment by girarda** [comment]
> Unrelated to this PR: @girarda, over the long term, please tell me we can use Python tooling to build and publish the CDK!
yes please @natikgadzhi

File: .github/workflows/publish-cdk-command-manually.yml [hunk 2]
```diff
@@ -73,7 +73,7 @@ jobs:
           repository: ${{ github.event.inputs.repo }}
           ref: ${{ github.event.inputs.gitref }}
       - name: Build CDK Package
-        run: ./gradlew --no-daemon --no-build-cache :airbyte-cdk:python:build
+        run: (cd airbyte-cdk/python; ./gradlew --no-daemon --no-build-cache :build)
```
[the remark this answers is not in the document]
Comment (postamar) [reply .github/workflows/publish-cdk-command-manually.yml:76]: `build` is `assemble`+`check`, we probably want the `check` in order to run the tests before publishing

File: airbyte-cdk/python/bin/build_code_generator_image.sh
[the remark this answers is not in the document]
Comment (postamar) [reply airbyte-cdk/python/bin/build_code_generator_image.sh:1]: yes indeed

**Comment by postamar** [comment]
Thanks for taking a look everyone! I'll merge this now.

**Pull request merged**